controllers/movies.ts

```diff
@@ -0,0 +1,171 @@
+import { error } from "console";
+import { Movie } from "../model/Movie";
+import { ILike } from "typeorm";
+
+export const getAllMovies= async( req:any, res: any, next:any)=>{
+
+    try{
+        
+        const { actor, imdbSort, yearSort  }= req.query;
+        let query= Movie.createQueryBuilder("movie");
+        if(actor){
+            query=query.where("movie.Actors LIKE :actor", {actor:`%${actor}%`});
+        } 
+        if(imdbSort){
+            const imdbSortDirection=imdbSort.toUpperCase()==="ASC" ? "ASC" :"DESC";
+            query= query.orderBy("movie.imdbRating", imdbSortDirection);
+        }
+
+        if(yearSort){
+            const yearSortDirection=yearSort.toUpperCase()==="ASC" ? "ASC" :"DESC";
+            query= query.orderBy("movie.Year", yearSortDirection);
+        }
+        let movies;
+        if(imdbSort){
+            movies= await query.select(["movie.Title", "movie.imdbRating"]).getMany();
+        }
+        else{
+            movies= await query.getMany();
+        }
+        //console.log(movies);
+        res.status(200).json({ movies: movies})
+
+    }catch(err:any){
+        //console.log("error")
+        err.message='Movies not found';
+        res.status(404).json({message: err.message})
+
+    }
+}
+
+export const getMovieByImdID= async(req:any, res:any, next:any)=>{
+    const imdb= req.params.imdb;
+    try{
+        const movie= await Movie.findOne({
+            where:{
+                imdbID:imdb
+            }
+        });
+
+        res.status(200).json({movie:movie})
+
+    }catch(err: any){
+        //console.log("error")
+        err.message='Movies not found';
+        res.status(404).json({message: err.message})
+
+    }
+}
+
+export const deleteMovie= async(req:any, res:any, next:any)=>{
+    const id= req.params.id;
+    try{
+        const movie= await Movie.findOne({
+            where:{
+                id:id
```

Comment:
Shorthand syntax. If the property has the same name and type you can write it like this: id, instaed of id:id,